docs/components/COMPONENTS.md

```diff
@@ -0,0 +1,38 @@
+# Component Folder and File Hierarchy
+
+## Sections
+
+- _Each component is broken into 3 main code sources_
+
+1. Source Code
+    - The source code is located in `./components/$COMPONENT_DESIGNATOR/` for each component and is completely owned and controlled by FSAE Concordia
+    - The source code, while it varies between component, has these mandatory and typical areas:
+        1. **Mandatory** `./components/$COMPONENT_DESIGNATOR/build/`
+            - Contains all object files, binary files, etc...
+        2. `./components/$COMPONENT_DESIGNATOR/generated/`
+            - Contains code generated by scripts and/or programs which is used by the embedded system
+        3. **Mandatory** `./components/$COMPONENT_DESIGNATOR/HW/`
+            - Contains all firmware source code and header files which interfaces between System Calls and the Hardware
+        4. **Mandatory** `./components/$COMPONENT_DESIGNATOR/include/`
+            - Contains all System and Application header files
+        5. `./components/$COMPONENT_DESIGNATOR/lib/`
+            - Contains any library source code and header files which the System uses
+        6. **Mandatory** `./components/$COMPONENT_DESIGNATOR/RTOS`
+            -Contains all source code and header files relating to the implementation of the System's RTOS
+        7. **Mandatory** `./components/$COMPONENT_DESIGNATOR/src`
+            - Contains all System and Application level source code
```

Comment:
None of these are mandatory, components can have different structures if it works better.